### ISSUE TITLE
Consider options for templates that use `aws cloudformation package`

### ISSUE THREAD

**Issue by spockNinja** [awslabs/cfn-python-lint#346]
*cfn-lint version: `0.3.5`*

*When linting files that are going to pass through `aws cloudformation package`, some properties are considered invalid by the linter.*

https://docs.aws.amazon.com/cli/latest/reference/cloudformation/package.html

We use the `package` command to simplify our build process. Specifically for lambda functions, we use the format of:

```yaml
  LambdaFunction:
    Type: AWS::Lambda::Function
    Properties:
      Code: path/to/local/folder
      Handler: handler.handle
      Runtime: python3.6
```

This format generates an error like:
```
E3002 Expecting an object at Resources/LambdaFunction/Properties/Code
```

This is then packaged up and the `Code` property is transformed. I think that the packaged version would then pass the linter.

However, we need to be able to lint in our IDEs and in our CI before we get to the build process that generates the s3 artifacts.

I thought about ignoring that error code, but is a very generic error code that would cover up way too many other errors.

Is there an existing option that I have not considered?

If not, would a PR be welcome to accommodate this use case? I have not yet dug into the source code, so I do not yet have a proposal for what this change would look like just yet.

Thanks!

**Comment by spockNinja** [comment]
Sorry, I just needed to upgrade! Looks like you've tackled this and added `W3002`. Sorry to bother! Thanks for the great work!
